internal/pkg/pool/pool.go

```diff
@@ -0,0 +1,123 @@
+// Package pool provides reusing of buffered and GZIP readers and writers to optimize memory usage.
+package pool
+
+import (
+	"bufio"
+	"io"
+	"runtime"
+	"sync"
+
+	"github.com/c2h5oh/datasize"
+	"github.com/klauspost/pgzip"
+)
+
+type BufferWriterPool struct {
+	pool *sync.Pool
+}
+
+type GZIPReaderPool struct {
+	pool *sync.Pool
+}
+
+type GZIPWriterPool struct {
+	pool *sync.Pool
+}
+
```

Comment:
So split processor bol problem s memory overflow, je tam viac dovodov, vid dalsie PRs.

Jeden problem je, ze sa pre kazdu `slice` vytvaraju nove buffers a GZIP readers/writers.
- To sposobje zahodenie starej memory, a alokovanie novej memory (uplne zbytocne).

Preto som pridal `pool` pkg, ktory je postaveny nad STD `sync.Pool`.

Ked vratis "nieco" spat do `pool`, cez `put`, tak sa to moze znova pouzit (zavola sa `Reset`).